sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheck.java

```diff
@@ -0,0 +1,246 @@
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.AnnotationUtility;
+
+/**
+ * <p>
+ * Checks if a try/catch block has a junit fail assertion inside the try for a junit method.
+ * </p>
+ * <p>
+ * Rationale: Tests should not complete the try block naturally if they are expecting a failure.
+ * If the try completes normally the test will pass successfully and skip over any assertions in
+ * the catch block.
+ * If tests are not expecting exceptions, then they should remove the catch block and propagate
+ * the exception to the junit caller which will display the full exception to the user.
+ * </p>
+ * <p>
+ * A junit test method is identified by the annotations placed on it. It is only considered a junit
+ * method if it contains the annotation 'org.junit.Test'. This check doesn't examine methods called
+ * by a test method. It must contain the annotation. Failures are identified by the
+ * method call to the method 'org.junit.Assert.fail'.
+ * </p>
+ * <p>
+ * An example of how to configure the check is:
+ * </p>
+ * <pre>
+ * &lt;module name="RequireFailForTryCatchInJunitCheck"/&gt;
+ * </pre>
+ * <p>
+ * which will cause a violation in the example below:
+ * </p>
+ * <pre>
+ *   &#064;Test
+ *   public void testMyCase() {
+ *     try { // violation here as try block has no 'Assert.fail()'.
+ *       verifySomeResult();
+ *     }
+ *     catch (IllegalArgumentException ex) {
+ *       assertEquals("expected exception message",
+ *           "Some message that is expected", ex.getMessage());
+ *     }
+ *   }
+ * </pre>
+ * @author Richard Veach
+ */
+public class RequireFailForTryCatchInJunitCheck extends AbstractCheck {
+    /**
+     * Violation message key.
+     */
+    public static final String MSG_KEY = "require.fail";
+
+    /**
+     * Fully qualified junit test annotation.
+     */
+    private static final String FQ_JUNIT_TEST = "org.junit.Test";
+    /**
+     * JUnit's fail assertion method name.
+     */
+    private static final String FAIL = "fail";
+
+    /**
+     * {@code true} if the junit test is imported.
+     */
+    private boolean importTest;
+    /**
+     * {@code true} if the junit assert is imported.
+     */
+    private boolean importAssert;
+    /**
+     * {@code true} if the junit fail assertion method is statically imported.
+     */
+    private boolean importStaticFail;
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.IMPORT,
+            TokenTypes.STATIC_IMPORT,
+            TokenTypes.LITERAL_TRY,
+        };
```

Comment:
declaration of such tokens should be at getRequiredTokens.

Reply:
Done.